### ISSUE TITLE
Add `shiny deploy` to CLI

### ISSUE THREAD

**Issue by nealrichardson** [rstudio/py-shiny#521]
I made a fun little app that I want to share. I want `shiny deploy my-app` to just work. Something along the lines of [`quarto publish`](https://quarto.org/docs/publishing/). I'd like to have the complexity of configuring rsconnect-python, connecting to shinyapps.io or Posit Connect, or even bundling and pushing to shinylive to be handled for me as much as possible.

**Comment by GShotwell** [comment]
I liked that idea, I'm a bit worried about recommending multiple deployment workflows. We do need to keep the `rsconnect` one because that's how people deploy other python applications to Posit Connect and Cloud. Can you say more about the complexity you found with `rsconnect`, maybe we can improve that package and have `shiny deploy` just be a pass through.

**Comment by nealrichardson** [comment]
Right, I think `shiny deploy` would wrap `rsconnect-python` to deploy to the targets that use it.

> Can you say more about the complexity you found with `rsconnect`

I started reading the docs and gave up without trying. Not to say that I won't give it a try later--I'm just wearing my impatient new user hat on, and when I run into friction, I take it as a sign that we need to reduce friction.

**Comment by nealrichardson** [comment]
@wch that's nice, but that only works if I'm already at shinylive.io developing. Suppose `shiny deploy shinylive` would generate that link? (is it just b64encoding the app file(s)?)

**Comment by wch** [comment]
@nealrichardson I can imagine a workflow like this:

- Edit app in VS Code
- Click on a Play button to try to run the app on shinylive.io. It could also support autoreload on save.
- Click on a Share button on the shinylive.io page to get the sharing URL.

Or, without VS Code:
- Edit app in some editor
- Run `shinylive watch .` to open the app on shinylive.io and autoreload on save.
- Click on a Share button on the shinylive.io page to get the sharing URL.

Other commands for `shinylive` could be:
- `shinylive url myapp`: Get share URL for app
- `shinylive open myapp`:  Open app in shinylive.io editor page. (Maybe have an option for opening without the editor)


I'm writing this as though these would use the `shinylive` command from the Python [shinylive](https://github.com/rstudio/py-shinylive) package. I think it conceptually makes sense to have the command be there, but on the other hand, that's one more thing the user would have to install, and if it were instead a `shiny` command, that would remove one step that the user would need to do.


*****

I think the CLI command is a good idea, but one more thing that might be useful for you now: the shinylive.io editor page has a button to load all files from a directory on disk (works in Chrome only, I think).

**Comment by nealrichardson** [comment]
> I'm writing this as though these would use the `shinylive` command from the Python [shinylive](https://github.com/rstudio/py-shinylive) package. I think it conceptually makes sense to have the command be there, but on the other hand, that's one more thing the user would have to install

We could put it in shinylive, have shiny depend on shinylive (or does that create a circular dependency problem?), and have `shiny deploy shinylive` call out to shinylive to render?

**Comment by gadenbuie** [comment]
Two recent updates:

1. `shinylive` now supports "deploying" an app to shinylive.io by creating a shinylive.io URL from local files. The command is `shinylive url encode`.
2. The Shiny VS Code extension also supports deploying to shinylive with two commands
	1. For single-file apps, e.g. `app.py` or `app.R`, run the **Create ShinyLive Link from Active File** from the command palette with the app file open and active.
	2. For multi-file apps, select all of the files or directories in the Explorer pane that you want to include in your Shinylive app. Then right click on the selection and choose **Create ShinyLive Link from Selected Files**.
